### PR TITLE
Update firebase-messaging version to 23.4.0

### PR DESCRIPTION
**Pull request by jinliu9508**
# Description
## One Line Summary
Update prefer version of firebase-messaging to 23.4.0

### Motivation
Several users have reported that updating the version of firebase-messaging resolves the 'Subscription can't change notification_types to positive integer while disabled' issue. Although a more precise cause is still under investigation, this change doesn't seem to break any existing functionality.

# Testing
## Manual testing
I am unable to reproduce the issue, as it seems to occur only for certain users, and the environment data is unclear at this moment.
I have tested push notification sent from the portal, push notification with the app closed, push notification on new install, and push notification click. The test was conducted on pixel 7 emulator with api 33. Everything works as expected. 

# Affected code checklist
   - [ ] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [ ] REST API requests
   - [ ] Public API changes

# Checklist
## Overview
   - [x] I have filled out all **REQUIRED** sections above
   - [x] PR does one thing
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR
   - [x] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [ ] I have included test coverage for these changes, or explained why they are not needed
   - [ ] All automated tests pass, or I explained why that is not possible
   - [x] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [x] Code is as readable as possible.
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
   - [x] I have reviewed this PR myself, ensuring it meets each checklist item
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Android-SDK/1987)

### CODE REVIEW

File: OneSignalSDK/onesignal/notifications/build.gradle [hunk 1]
```diff
@@ -74,8 +74,8 @@ dependencies {
     // firebase-messaging:23.0.0 incoporates fix for SecurityException: Not allowed to bind to service
     api('com.google.firebase:firebase-messaging') {
         version {
-            require '[19.0.0, 23.0.99]'
-            prefer '23.0.8'
+            require '[19.0.0, 23.99.99]'
```
Comment (jkasten2) [inline OneSignalSDK/onesignal/notifications/build.gradle:77]: I would recommend bumping only the minor version number in this version range. We have seen Firebase and Google break things in minor version bumps in the past.  So thinking this should be `[19.0.0, 23.4.99]`.

**Review by jkasten2 — CHANGES_REQUESTED** [review]
Left a comment on the `require` version range.

Also can you note what was tested and on what device / android version in the PR?
* Testing registering on a new install and that notifications display is normally all we need to test when updating the FCM library.

File: OneSignalSDK/onesignal/notifications/build.gradle (hunk 1 above)
Comment (jinliu9508) [reply OneSignalSDK/onesignal/notifications/build.gradle]: That makes sense, I pushed the change just now.

**Comment by jinliu9508** [comment]
> Left a comment on the `require` version range.
> 
> Also can you note what was tested and on what device / android version in the PR?
> 
> * Testing registering on a new install and that notifications display is normally all we need to test when updating the FCM library.

Ok, I have updated the PR description with the testing environment and the testing units I have done for this PR.

**Review by jkasten2 — APPROVED** [review]

**Pull request merged**